### PR TITLE
SW-1797 Allow changing status from Used Up to any status

### PR DESCRIPTION
**Pull request by karthikbtf**
- require quantity edits at this point
- refactored code to reuse some components and modals

Terraware App 2022-09-28 16-41-49

### CODE REVIEW

File: src/components/accession2/edit/EditState.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
```
Comment (karthikbtf) [inline src/components/accession2/edit/EditState.tsx:1]: this was extracted out from EditStateModal... and it now also instantiated in quantity modal when we edit Used Up status.

**Comment by constanzauanini** [comment]
How can I end with a Used up status? I couldn't test this

**Comment by karthikbtf** [comment]
> How can I end with a Used up status? I couldn't test this

Withdraw All

**Comment by constanzauanini** [comment]
> > How can I end with a Used up status? I couldn't test this
> 
> Withdraw All

Thanks! Looking good. Not related to this, I don't see the pencil to edit quantity appearing on hover

**Review by constanzauanini — APPROVED** [review]

**Comment by constanzauanini** [comment]
> > > How can I end with a Used up status? I couldn't test this
> > 
> > 
> > Withdraw All
> 
> Thanks! Looking good. Not related to this, I don't see the pencil to edit quantity appearing on hover

Oh, ignore my previous comment. I forgot it has to be drying or in storage!

**Pull request merged**